src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -0,0 +1,1088 @@
+                        {
+                            // ParentId is only used if there is a memberInfo
+                            memberInfo = objectPr._objectInfo.GetMemberInfo(pr._name);
+                        }
+                    }
+                }
+                // SerializationInfo is always needed for ISerialization                        
+                si = pr._si;
+
+                if (bIsString)
+                {
+                    _objectManager.RegisterString((string)obj, pr._objectId, si, parentId, memberInfo);
+                }
+                else
+                {
+                    _objectManager.RegisterObject(obj, pr._objectId, si, parentId, memberInfo, indexMap);
+                }
+            }
+        }
+
+        // Assigns an internal ID associated with the binary id number
+        internal long GetId(long objectId)
+        {
+            if (!_fullDeserialization)
+            {
+                InitFullDeserialization();
+            }
+
+            if (objectId > 0)
+            {
+                return objectId;
+            }
+
+            if (_oldFormatDetected || objectId == -1)
+            {
+                // Alarm bells. This is an old format. Deal with it.
+                _oldFormatDetected = true;
+                if (_valTypeObjectIdTable == null)
+                {
+                    _valTypeObjectIdTable = new IntSizedArray();
+                }
+
+                long tempObjId = 0;
+                if ((tempObjId = _valTypeObjectIdTable[(int)objectId]) == 0)
+                {
+                    tempObjId = ThresholdForValueTypeIds + objectId;
+                    _valTypeObjectIdTable[(int)objectId] = (int)tempObjId;
+                }
+                return tempObjId;
+            }
+
+            return -1 * objectId;
+        }
+
+        internal Type Bind(string assemblyString, string typeString)
+        {
+            Type type = null;
+            if (_binder != null)
+            {
+                type = _binder.BindToType(assemblyString, typeString);
+            }
+            if (type == null)
+            {
+                type = FastBindToType(assemblyString, typeString);
+            }
+            return type;
+        }
+
+        internal sealed class TypeNAssembly
+        {
+            public Type Type;
+            public string AssemblyName;
+        }
+
+        internal Type FastBindToType(string assemblyName, string typeName)
+        {
+            Type type = null;
+
+            TypeNAssembly entry = (TypeNAssembly)_typeCache.GetCachedValue(typeName);
+
+            if (entry == null || entry.AssemblyName != assemblyName)
+            {
+                Assembly assm = null;
+                if (_isSimpleAssembly)
+                {
+                    try
+                    {
+                        Assembly.Load(new AssemblyName(assemblyName));
+                    }
+                    catch { }
```

Comment:
> So there is no semantic difference between catch{} and catch(Exception) {}, unless you override the default for WrapNonExceptionThrows.

Clearly a thing I didn't know until now.  Thanks, @jkotas.